### DISCUSSION TITLE
Consistent library module name

### DISCUSSION THREAD

**Discussion by wsmelton** [dataplat/dbatools#8715] [Contributors]
I would much rather we keep the same name of the library module no matter what version of PowerShell is in use.

https://github.com/dataplat/dbatools/blob/cb554f576758b915252a2a5b853aea9ac443b6dc/dbatools.psd1#L52-L62

It will lessen the burden on our sanity with maintaining and documentation for the project.

![image](https://user-images.githubusercontent.com/11204251/211170325-6b1f80c8-8b01-4c12-b2d1-eb26998162e8.png)

**Comment by potatoqualitee** [comment]
Unfortunately, I do not think this is possible. Knowing exactly where a DLL will be no matter the version makes it 100x easier while loading. In addition, this will increase the size of the module and also presents problems with the way the DLL loads and we'd be back to the old way of fighting SMO.

**Reply by wsmelton** [reply]
Be aware at this time the only way users can discover the `dbatools.library` is by using the prerelease flag for `Install-Module` or `Install-PSResource`. Right now just installing with out that flag is installing the core library on any version of PowerShell.

**Reply by potatoqualitee** [reply]
If people want it, they can have it 😅  I was testing prerelease on one of the modules. they are both prerelease now. install-module should only install the required ps module eh? otherwise that's trash.

**Reply by potatoqualitee** [reply]
No way can this be true 😡 

![image](https://user-images.githubusercontent.com/8278033/215293775-3aef0435-9e63-4f2d-b6b9-b4c03b20a574.png)

**Reply by wsmelton** [reply]
Assuming you are referring to the fact that pre-release won't install other dependent modules that are also pre-release? If so no, it never has to my knowledge. Users will need to do install command for both.

I'm not sure if they address that in the PSGet v3 or not yet.

**Reply by potatoqualitee** [reply]
OMGGGGGGGGGGGGGGGGGGGGGGGGGGGG THIS IS BEYOND RIDICULOUS AND I AM GOBSMACKED

it didnt let me publish with prerelease libraries so i did full release on the libraries and a prerelease on dbatools

I gotta go undo all that effin work. i.. i am in disbelief. WHY INTRODUCE IF PSCORE IF IT'S NOT SUPPORTED 😡 😡 😡 😡 😡 😡

**Reply by potatoqualitee** [reply]
I'll see if i can move the burden to Get-DbatoolsLibraryPath 😡

**Reply by wsmelton** [reply]
PowerShell Get v2 has this issue since 2019 powershell/powershellgetv2#595

I don’t find an issue matching on [PowerShellGet repo](https://github.com/powershell/powershellget)

**Reply by wsmelton** [reply]
Do we want a command being run that will install additional bits not sourced in the original module automatically?

**Reply by potatoqualitee** [reply]
The get-dbatoolslibrarypath made it way easier, so i just undid it all and unpublished a few of the modules from the gallery. thanks for helping to clarify what was happening.

https://github.com/dataplat/dbatools/pull/8741

the downside is that the library is twice as big, but the upside is that this will simplify everything and keep the book relevant. i'm very surprised and disappointed that this implementation was so half baked. i dont remember if i did this in any of my other modules, but I'll need to go undo it if so 😱

**Comment by potatoqualitee** [comment]
this still shocks me

**Reply by potatoqualitee** [reply]
wrote about it here https://blog.netnerds.net/2023/03/dont-waste-your-time-with-core-versions/
